server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/Routes.kt

```diff
@@ -5,65 +5,124 @@ import com.aallam.openai.api.chat.ChatCompletionRequest
 import com.xebia.functional.xef.conversation.Conversation
 import com.xebia.functional.xef.prompt.configuration.PromptConfiguration
 import com.xebia.functional.xef.conversation.llm.openai.*
+import com.xebia.functional.xef.llm.StreamedFunction
+import com.xebia.functional.xef.llm.models.chat.ChatCompletionRequest as XefChatCompletionRequest
+import com.xebia.functional.xef.llm.models.chat.ChatCompletionResponse
 import com.xebia.functional.xef.prompt.Prompt
 import com.xebia.functional.xef.server.services.PersistenceService
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
 import io.ktor.http.*
 import io.ktor.server.application.*
 import io.ktor.server.auth.*
 import io.ktor.server.request.*
 import io.ktor.server.response.*
 import io.ktor.server.routing.*
+import io.ktor.util.cio.*
 import io.ktor.util.pipeline.*
+import io.ktor.utils.io.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.boolean
+import kotlinx.serialization.json.jsonPrimitive
 
 enum class Provider {
-  OPENAI, GPT4ALL, GCP
+    OPENAI, GPT4ALL, GCP
 }
 
 fun String.toProvider(): Provider? = when (this) {
-  "openai" -> Provider.OPENAI
-  "gpt4all" -> Provider.GPT4ALL
-  "gcp" -> Provider.GCP
-  else -> null
+    "openai" -> Provider.OPENAI
+    "gpt4all" -> Provider.GPT4ALL
+    "gcp" -> Provider.GCP
+    else -> null
 }
 
 
 @OptIn(BetaOpenAI::class)
-fun Routing.routes(persistenceService: PersistenceService) {
-  authenticate("auth-bearer") {
-    post("/chat/completions") {
-      val provider: Provider = call.request.headers["xef-provider"]?.toProvider()
-        ?: throw IllegalArgumentException("Not a valid provider")
-      val token = call.principal<UserIdPrincipal>()?.name ?: throw IllegalArgumentException("No token found")
-      val scope = Conversation(
-        persistenceService.getVectorStore(provider, token)
-      )
-      val data = call.receive<ChatCompletionRequest>().toCore()
-      val model: OpenAIModel = data.model.toOpenAIModel(token)
-      response<String, Throwable> {
-        model.promptMessage(
-          prompt = Prompt(
-            messages = data.messages,
-            configuration = PromptConfiguration(
-              temperature = data.temperature,
-              numberOfPredictions = data.n,
-              user = data.user ?: ""
-            )
-          ),
-          scope = scope
-        )
-      }
+fun Routing.routes(
+    client: HttpClient,
+    persistenceService: PersistenceService
+) {
+    val openAiUrl = "https://api.openai.com/v1"
+
+    authenticate("auth-bearer") {
+        post("/chat/completions") {
+            val provider: Provider = call.getProvider()
```

Comment:
At the moment this endpoint is only for OpenAI, I think that we don't need to send the `xef-provider` header in the call